src/docs/release/breaking-changes/text-field-material-localizations.md

```diff
@@ -0,0 +1,140 @@
+---
+title: TextField requires Material Localizations
+description: TextFields will now assert for Material Localizations.
+---
+
+## Summary
+
+Instances of `TextField` must have a `MaterialLocalizations` present.
+Trying to instantiate  a `TextField` without the proper localizations will
+result in an assertion such as the following:
+
+```
+No MaterialLocalizations found.
+TextField widgets require MaterialLocalizations to be provided by a Localizations widget ancestor.
+Localizations are used to generate many different messages, labels, and abbreviations which are used
+by the material library.
+To introduce a MaterialLocalizations, either use a MaterialApp at the root of your application to
+include them automatically, or add a Localization widget with a MaterialLocalizations delegate.
+The specific widget that could not find a MaterialLocalizations ancestor was:
+  TextField
+```
+
+## Context
+
+If your `TextField` is within the scope of a `MaterialApp`, the
+`DefaultMaterialLocalizations` will already be instantiated and may suit your needs without having
+to make a change to your existing code.
+
+If your `TextField` is not within a `MaterialApp`, you may provide the localizations of
+your choosing using the `Localizations` widget.
+
+This change will initially be introduced with an opt-in flag
+`TextField.canAssertMaterialLocalizations`, to allow developers time to migrate
+without experiencing broken code. This temporary opt-in flag will be removed in a follow up change.
```

Comment:
In a later release? Do we know when?

Reply:
I don't actually know... According to the breaking change policy it does not specify a timeline for these opt-in changes. https://github.com/flutter/flutter/wiki/Tree-hygiene#3-prepare-your-change @Hixie do you know how long we let these opt-in flags marinate?

Reply:
we don't. we only have them to help us land. we shouldn't mention them in these docs.

the only exception would be if you actually thought it valuable to have them for a long time, in which case we should decide either to have them forever, or to deprecate them immediately (after we've landed), which will add them to our deprecation pipeline. At some point we'll start cleaning up deprecated stuff. (If you are interested in doing that let me know. I've set things up so that deprecations say when they were added, so we can use that to apply a moving sweep that collects deprecations that are older than some set duration.)